examples/image_classification/ios/ImageClassifier/ViewContoller/ViewController.swift

```diff
@@ -0,0 +1,400 @@
+                   object: player.currentItem
+      )
+
+    videoDetectTimer?.invalidate()
+    videoDetectTimer = Timer.scheduledTimer(
+      timeInterval: delayBetweenInferencesMs / 1000,
+      target: self,
+      selector: #selector(classificationVideoFrame),
+      userInfo: nil,
+      repeats: true)
+  }
+
+  @objc func classificationVideoFrame() {
+    guard let player = playerViewController.player else { return }
+    let currentTime: CMTime = player.currentTime()
+    guard let buffer = self.pixelBufferFromCurrentPlayer(currentTime: currentTime) else { return }
+    let currentTimeMs = Date().timeIntervalSince1970 * 1000
+    let result = self.imageClassifierHelper?.classify(videoFrame: buffer, timeStamps: Int(currentTimeMs))
+    // Display results by handing off to the InferenceViewController.
+    inferenceViewController?.imageClassifierHelperResult = result
+    DispatchQueue.main.async {
+      self.inferenceViewController?.updateData()
+    }
+  }
+
+  @objc func playerDidFinishPlaying(note: NSNotification) {
+    videoDetectTimer?.invalidate()
+    videoDetectTimer = nil
+  }
+  private func pixelBufferFromCurrentPlayer(currentTime: CMTime) -> CVPixelBuffer? {
+    guard let buffer: CVPixelBuffer = videoOutput.copyPixelBuffer(forItemTime: currentTime, itemTimeForDisplay: nil) else { return nil }
+    return buffer
+  }
+}
+
+// MARK: UIImagePickerControllerDelegate, UINavigationControllerDelegate
+extension ViewController: UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+  func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
+    picker.dismiss(animated: true)
+    if info[.mediaType] as? String == UTType.movie.identifier {
+      guard let mediaURL = info[.mediaURL] as? URL else { return }
+      imageEmptyLabel.isHidden = true
+      if runingModel != .video {
+        runingModel = .video
+      }
+      processVideo(url: mediaURL)
+    } else {
+      guard let image = info[.originalImage] as? UIImage else { return }
+      if runingModel != .image {
+        runingModel = .image
+      }
+      removePlayerViewController()
+      previewView.image = image
+      imageEmptyLabel.isHidden = true
+      // Pass the uiimage to mediapipe
+      let result = imageClassifierHelper?.classify(image: image)
+      // Display results by handing off to the InferenceViewController.
+      inferenceViewController?.imageClassifierHelperResult = result
+      DispatchQueue.main.async {
+        self.inferenceViewController?.updateData()
+      }
+    }
+  }
+}
+
+// MARK: CameraFeedManagerDelegate Methods
+extension ViewController: CameraFeedManagerDelegate {
+
+  func didOutput(pixelBuffer: CVPixelBuffer) {
+    // Make sure the model will not run too often, making the results changing quickly and hard to
+    // read.
+    let currentTimeMs = Date().timeIntervalSince1970 * 1000
```

Comment:
You don't have to throttle the inference code. Just pass all frames you receive from the camera to MediaPipe and it will drop frames that it can't process automatically. See the Android code for reference.
https://github.com/googlesamples/mediapipe/blob/39ce84ae02628d7891d8a6656c7e8de860d31df8/examples/image_classification/android/app/src/main/java/com/google/mediapipe/examples/imageclassification/ImageClassifierHelper.kt#L183